classes/dependency-track.bbclass

```diff
@@ -53,11 +58,15 @@ python do_dependencytrack_collect() {
     for index, cpe in enumerate(oe.cve_check.get_cpe_ids(name, version)):
         bb.debug(2, f"Collecting pagkage {name}@{version} ({cpe})")
         if not next((c for c in sbom["components"] if c["cpe"] == cpe), None):
-            sbom["components"].append({
+            component_json = {
                 "name": names[index],
                 "version": version,
-                "cpe": cpe
-            })
+                "cpe": cpe,
+            }
+            license_json = get_licenses(d)
+            if license_json:
+                component_json["licenses"] = license_json
```

Comment:
This line gives a structure like 
"licenses": [
        {
          "license": {

according to 

https://cyclonedx.org/use-cases/#license-compliance

Can you check if you have this structure in the SBOM?

Reply:
it's in the right structure, or I don't see the failure myself. But dependency-track still don't show the licenses if I upload the sbom to the api.

Reply:
I tried this:
    # update it with the new package info
    names = name.split()
    for index, cpe in enumerate(oe.cve_check.get_cpe_ids(name, version)):
        bb.debug(2, f"Collecting pagkage {name}@{version} ({cpe})")
        if not next((c for c in sbom["components"] if c["cpe"] == cpe), None):
            sbom["components"].append({
                "name": names[index],
                "version": version,
                "cpe": cpe,
                "licenses" : [
                {
                "license" : {
                "id" : license
                  }
                }]
            })

before your last update and it shows like 50% of the licenses.